ibis/expr/datatypes/__init__.py

```diff
@@ -82,11 +85,17 @@ def _factory(self, nullable: bool = True) -> DataType:
         }
         return self.__class__(nullable=nullable, **slots)
 
-    def __eq__(self, other) -> bool:
-        return self.equals(other)
-
-    def __ne__(self, other) -> bool:
-        return not (self == other)
+    def equals(
+        self,
+        other: typing.Any,
+        cache: MutableMapping[Hashable, bool] | None = None,
+    ) -> bool:
+        if not isinstance(other, DataType):
+            raise TypeError(
+                'Comparing datatypes to other types is not allowed. Convert '
```

Comment:
shouldn't this return False? and not raise

Reply:
No, comparing a datatype with something other than a datatype is an error.